core/lib/src/config/secret_key.rs

```diff
@@ -178,6 +187,75 @@ impl SecretKey {
     {
         ser.serialize_bytes(&[0; 32][..])
     }
+
+    /// Encrypts the given plaintext.
+    /// Generates a random nonce for each encryption to ensure uniqueness.
+    /// Returns the base64-encoded string of the concatenated nonce and ciphertext.
+    ///
+    /// # Example
+    /// ```rust
+    /// let plaintext = "I like turtles";
+    /// let secret_key = SecretKey::generate().expect("error generate key");
+    ///
+    /// let encrypted = secret_key.encrypt(&plaintext).expect("can't encrypt");
+    /// let decrypted = secret_key.decrypt(&encrypted).expect("can't decrypt");
+    ///
+    /// assert_eq!(decrypted, plaintext);
+    /// ```
+    pub fn encrypt(&self, plaintext: &str) -> Result<String, &'static str> {
```

Comment:
Should we restrict this to only encrypting strings? It wouldn't be hard to offer this for any `impl AsRef<[u8]>`.

Obviously, we would also need to implement a decrypt that decrypts to bytes, since it might not be a UTF-8 string.

Also, `Result<_, &'static str>` isn't very good practice. We should instead create an error enum, and return that.